pandas/core/arrays/base.py

```diff
@@ -653,15 +655,65 @@ def copy(self, deep=False):
         raise AbstractMethodError(self)
 
     # ------------------------------------------------------------------------
-    # Block-related methods
+    # Printing
     # ------------------------------------------------------------------------
+    def __repr__(self):
+        from pandas.io.formats.printing import format_object_summary
+
+        template = (
+            u'{class_name}'
+            u'{data}\n'
+            u'Length: {length}, dtype: {dtype}'
+        )
+        # the short repr has no trailing newline, while the truncated
+        # repr does. So we include a newline in our template, and strip
+        # any trailing newlines from format_object_summary
+        data = format_object_summary(self, self._formatter(), name=False,
+                                     trailing_comma=False).rstrip()
+        class_name = u'<{}>\n'.format(self.__class__.__name__)
+        return template.format(class_name=class_name, data=data,
+                               length=len(self),
+                               dtype=self.dtype)
+
+    def _formatter(self, boxed=False):
+        # type: (bool) -> Callable[[Any], str]
+        """Formatting function for scalar values.
+
+        This is used in the default '__repr__'. The returned formatting
+        function receives instances of your scalar type.
+
+        Parameters
+        ----------
+        boxed: bool, default False
```

Comment:
We needed it to get the period tests passing. PeriodIndex quotes scalar values, but `Series[Period]` doesn't (which is fine, since the Series repr isn't supposed to be code).

Reply:
well, maybe fix that. This is completely inconsistent with the rest of the reprs.

Reply:
Datetime is the same.

```python
In [2]: pd.date_range('2017', periods=4)
Out[2]: DatetimeIndex(['2017-01-01', '2017-01-02', '2017-01-03', '2017-01-04'], dtype='datetime64[ns]', freq='D')

In [3]: pd.Series(pd.date_range('2017', periods=4))
Out[3]:
0   2017-01-01
1   2017-01-02
2   2017-01-03
3   2017-01-04
dtype: datetime64[ns]
```

Reply:
this is a Series repr, that is different from Index. Array is *much* more like Index.

Reply:
Responded in 
https://github.com/pandas-dev/pandas/pull/23601#issuecomment-439882626